### PR TITLE
Av 1052 high g accelerometer implementation

### PR DESCRIPTION
**Pull request by SuragNuthulapaty**
Added high g data functionality based on tars

### CODE REVIEW

File: MIDAS/src/errors.h [hunk 1]
```diff
@@ -3,5 +3,6 @@
 
 enum ErrorCode {
     NoError,
-    LowGCouldNotBeInitialized
+    LowGCouldNotBeInitialized,
+    CANNOT_INIT_KX134_CS,
```
Comment (aidancostello) [inline MIDAS/src/errors.h:7]: Try to copy the same format as LowGCouldNotBeInitialized. Also did you mean to leave the cs in there

File: MIDAS/src/systems.cpp [hunk 2]
```diff
@@ -36,6 +36,7 @@ DECLARE_THREAD(low_g, RocketSystems* arg) {
 
 DECLARE_THREAD(high_g, RocketSystems* arg) {
     while (true) {
+        arg->rocket_state.high_g.update(arg->sensors.high_g.read());
         THREAD_SLEEP(16);
```
Comment (aidancostello) [inline MIDAS/src/systems.cpp:40]: try to match the max sampling rate of the hardware

**Review by redindelible — CHANGES_REQUESTED** [review]
Yeah the Error name

**Comment by redindelible** [comment]
You need to merge with master I think

File: MIDAS/src/hardware/HighG.cpp [hunk 3]
```diff
@@ -1,17 +1,30 @@
...
 
+/**
+ * Reads and returns the data from the sensor
+ * @return a HighGData packet with current acceleration in all three axies
+*/
 HighGData HighGSensor::read() {
-    // read from aforementioned global instance of sensor
-    return HighGData();
+    auto data = KX.getAccelData();
+    return {.gx = data.xData, .gy = data.yData, .gz = data.zData};
```
Comment (aidancostello) [inline MIDAS/src/hardware/HighG.cpp:29]: errors in platformio build

File: MIDAS/src/systems.cpp [hunk 4]
```diff
@@ -47,7 +47,8 @@ DECLARE_THREAD(gyroscope, RocketSystems* arg) {
 
 DECLARE_THREAD(high_g, RocketSystems* arg) {
     while (true) {
-        THREAD_SLEEP(16);
+        arg->rocket_state.high_g.update(arg->sensors.high_g.read());
+        THREAD_SLEEP(20);
```
Comment (aidancostello) [inline MIDAS/src/systems.cpp:51]: match sampling rate as close as possible

**Review by aidancostello — CHANGES_REQUESTED** [review]

File: MIDAS/src/systems.cpp (hunk 4 above)
Comment (SuragNuthulapaty) [reply MIDAS/src/systems.cpp:51]: image

The data sheet i found [here](https://cdn.sparkfun.com/assets/6/a/2/c/b/KX134-1211-Specifications-Rev-1.0.pdf), on page 6 says typical sampling rate is 50 Hz. What should I change it to?

Comment (aidancostello) [reply MIDAS/src/systems.cpp:51]: whoops my bad you are correct :)

**Review by aidancostello — DISMISSED** [review]

**Pull request merged**